### ISSUE TITLE
The prometheus service account does not have enough Permissions

### ISSUE THREAD

**Issue by maryjomcguinness** [oracle/docker-images#779]
The issue can be seen when looking at the Prometheus logs

 **./kubectl logs prometheus-deployment-5cfdf8f756-mpctk --namespace=monitoring**
level=error ts=2018-03-06T15:18:52.981806942Z caller=main.go:221 component=k8s_client_runtime err="github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:268: Failed to list *v1.Endpoints: endpoints is forbidden: User \"system:serviceaccount:monitoring:default\" cannot list endpoints at the cluster scope"
level=error ts=2018-03-06T15:18:52.981851051Z caller=main.go:221 component=k8s_client_runtime err="github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:354: Failed to list *v1.Node: nodes is forbidden: User \"system:serviceaccount:monitoring:default\" cannot list nodes at the cluster scope"
level=error ts=2018-03-06T15:18:52.981878974Z caller=main.go:221 component=k8s_client_runtime err="github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:269: Failed to list *v1.Service: services is forbidden: User \"system:serviceaccount:monitoring:default\" cannot list services at the cluster scope"
level=error ts=2018-03-06T15:18:52.981907841Z caller=main.go:221 component=k8s_client_runtime err="github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:296: Failed to list *v1.Pod: pods is forbidden: User \"system:serviceaccount:monitoring:default\" cannot list pods at the cluster scope"

I seem to be getting the same errors above.
I tried to edit this file and just changed https://github.com/prometheus/prometheus/blob/master/documentation/examples/rbac-setup.yml
the namespace value to "monitoring" which is the namespace I am using.

I then tried to run the following command using that rbac-setup.yml file:

**./kubectl apply -f rbac-setup.yml**
clusterrole "prometheus" created
serviceaccount "prometheus" created
clusterrolebinding "prometheus" created

And checked the service account was there:

**./kubectl get serviceaccounts --namespace=monitoring**
NAME         SECRETS   AGE
default      1         5d
prometheus   1         24s

However I am still getting the same errors and there is nothing under metrics, targets or Service Discovery in the Prometheus UI. Could you please advise asap what else needs to be done?

**Comment by karan-kapoor90** [comment]
Yup, that's an issue. Check out the prometheus-kubernetes.yml https://github.com/karan-kapoor90/wls-K8s/blob/master/prometheus/prometheus-kubernetes.yml

I faced the same issue #770

**Comment by marinakog** [comment]
Hi @maryjomcguinness  , please let me know if this example solved your issue https://github.com/karan-kapoor90/wls-K8s/blob/master/prometheus/prometheus-kubernetes.yml

**Comment by mriccell** [comment]
@maryjomcguinness  are you still experiencing the same problem?
